### ISSUE TITLE
Incompatible return type (Deprecated)

### ISSUE THREAD

**Issue by oNdsen** [AsperaGmbH/xlsx-reader#14]
With PHP 8.1.7 im getting the following Deprecated Messages:
```
Deprecated:  Return type of Aspera\Spreadsheet\XLSX\Reader::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in dist\vendor\aspera\xlsx-reader\lib\Reader.php on line 244

Deprecated:  Return type of Aspera\Spreadsheet\XLSX\Reader::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in dist\vendor\aspera\xlsx-reader\lib\Reader.php on line 263

Deprecated:  Return type of Aspera\Spreadsheet\XLSX\Reader::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in dist\vendor\aspera\xlsx-reader\lib\Reader.php on line 485

Deprecated:  Return type of Aspera\Spreadsheet\XLSX\Reader::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in dist\vendor\aspera\xlsx-reader\lib\Reader.php on line 496

Deprecated:  Return type of Aspera\Spreadsheet\XLSX\Reader::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in dist\vendor\aspera\xlsx-reader\lib\Reader.php on line 219

Deprecated:  Return type of Aspera\Spreadsheet\XLSX\Reader::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in dist\vendor\aspera\xlsx-reader\lib\Reader.php on line 507
```

Im loading xlsx-reader via:
```php
use Aspera\Spreadsheet\XLSX\Reader;
use Aspera\Spreadsheet\XLSX\ReaderConfiguration;
use Aspera\Spreadsheet\XLSX\ReaderSkipConfiguration;

$options = (new ReaderConfiguration())
	->setSkipEmptyCells(ReaderSkipConfiguration::SKIP_EMPTY)
	->setReturnDateTimeObjects(true);
$reader = new Reader($options);
$reader->open($sheet);
```

Using the most current package:
`aspera/xlsx-reader                  v0.10.1 Spreadsheet reader library for Microsoft Excel XLSX files`

Is there something i miss?

Thanks, oNdsen

**Comment by rfischer42** [comment]
Greetings!

No, there's nothing you miss. : )

These deprecation notices were added in PHP 8.1, and support for PHP 8.1 wasn't checked for this library yet.
Given the information that's currently available to me, "complete" support for newer PHP versions will likely be added sometime within the next 4 weeks.

With that said: The important parts of this library should work fine in PHP 8.1.
The just-released version (v0.10.2) suppresses the notices you've been getting, as they are of no real concern yet.
As such, try updating to v0.10.2. Your logs should be quieter then. : )

Sincerely - René Fischer

**Comment by oNdsen** [comment]
Thanks René!

Yes, the Script worked also before without issues while surpressing Deprecated Messages. But i'd like to see, which functions are deprecated to avoid not working code after updates.

Thanks for the Fix, just tried and it looks great.

Best Regards
